### PR TITLE
Plan 61: Required Structure Rule Hardening (WIP)

### PR DESCRIPTION
**Pull request by jeduden**
Tracks work on [plan 61 — Required Structure Rule Hardening](../blob/main/plan/61_required-structure-hardening.md).

Picked plan 61 because it was the only unblocked plan with no existing branch or PR: 50, 52 are large new features; 87 depends on unimplemented plan 86 (MDS034).

## Changes by commit

1. **Plan bookkeeping** (`d3a8edc`): flip `status` to 🔳 in `plan/61_...md` and regenerate `PLAN.md`.
2. **Named diagnostics** (`7c42b38`):
   - Level-mismatch now names the offending heading: `heading level mismatch for "Settings": expected h2, got h3`.
   - Unexpected-section now names the required heading that was expected next: `unexpected section "## Extra" (expected "## Settings")`.
   - Trailing extras (past the last required heading) keep the original shorter phrasing.
3. **Out-of-order detection** (`0070c88`):
   - Document holding every required heading but in the wrong order previously produced both `unexpected section X` and `missing required section X` for the same heading.
   - Now emits a single `section X out of order: expected after Y` diagnostic; the schema index is claimed so no follow-up "missing" fires. Level mismatches on out-of-order matches are still reported.
   - Text-based matching excludes wildcard (`...`), `?`, and `{field}`-interpolated schema headings where out-of-order semantics would need fuzzy/context-aware matching.

Inner loop extracted to `matchRequired` to stay within the gocognit threshold. README diagnostic table updated with the new messages and an `out of order` row.

## Status

- [x] Mark plan as in-progress (🔳)
- [x] Expand unit tests with focused false-positive/negative fixtures *(level-mismatch naming, extra-section context, trailing extras, out-of-order)*
- [x] Refine matching logic for required headings *(out-of-order claim)*
- [x] Improve diagnostic messages (expected vs actual)
- [x] Update `MDS020-required-structure/README.md`
- [ ] Broader audit of sync-point edge cases and front-matter/body sync false positives — deferred to a follow-up PR.

## Test plan

- `go test ./...` — passes (unrelated `internal/corpus` git-signing failures only in the local sandbox; CI green).
- `go vet ./...` — clean.
- `go tool golangci-lint run` — 0 issues.
- `go run ./cmd/mdsmith check .` — 115 files checked, 0 failures.

### CODE REVIEW

File: plan/61_required-structure-hardening.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 id: 61
 title: Required Structure Rule Hardening
-status: 🔲
+status: 🔳
```
[the remark this answers is not in the document]
Comment (jeduden) [reply plan/61_required-structure-hardening.md:4]: Good catch — the description was stale. Updated the PR body to list all three commits (plan bookkeeping, named diagnostics, out-of-order detection) with scope per commit and a test plan. Keeping it in a single PR since the diagnostic and out-of-order changes are the actual plan 61 work; the bookkeeping commit just flips `status` so PLAN.md shows 🔳 while the work lands.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jeduden** [comment]
🟢 **Merge Queue** — picked up

This PR is in the queue and will be batched with other `queue`-labelled PRs.

**Next:** No action needed — you'll get another comment when CI starts on the batch. [View merge queue run](https://github.com/jeduden/mdsmith/actions/runs/24638790411).

**Comment by jeduden** [comment]
🔵 **Merge Queue** — CI running

Merged into batch branch [`merge-queue/batch-145-1776631701`](https://github.com/jeduden/mdsmith/tree/merge-queue/batch-145-1776631701). [View CI run](https://github.com/jeduden/mdsmith/actions/runs/24638793033).

**Next:** No action needed — you'll be notified when CI completes.

**Comment by jeduden** [comment]
✅ **Merge Queue** — merged

This PR landed on `main` via commit [`f48ec68`](https://github.com/jeduden/mdsmith/commit/f48ec6824d9e7fc1d64d94eaa3eb0c0046652937). [CI run that validated the merge](https://github.com/jeduden/mdsmith/actions/runs/24638793033).

**Next:** Done — nothing more to do here.
